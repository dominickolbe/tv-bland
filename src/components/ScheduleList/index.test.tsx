import { shallow } from "enzyme";
import React from "react";
import scheduleJSON from "../../__mock__/schedule.json";
import { ScheduleList } from "./";

describe("<ScheduleList />", () => {
  it("renders snapshots", () => {
    const wrapper = shallow(
      <ScheduleList
        // @ts-ignore
        schedule={scheduleJSON}
        onClickCard={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
