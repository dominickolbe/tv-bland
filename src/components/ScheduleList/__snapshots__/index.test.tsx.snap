// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduleList /> renders snapshots 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={3}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1937214"
    md={2}
    sm={4}
    xs={6}
  >
    <TvShowEpisodeCard
      onClick={[MockFunction]}
      scheduleItem={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/1937214",
            },
          },
          "airdate": "2020-10-25",
          "airstamp": "2020-10-25T12:30:00+00:00",
          "airtime": "08:30",
          "id": 1937214,
          "image": null,
          "name": "The Shawhank Redemption",
          "number": 41,
          "runtime": 15,
          "season": 2,
          "show": Object {
            "_links": Object {
              "nextepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1937216",
              },
              "previousepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1937215",
              },
              "self": Object {
                "href": "http://api.tvmaze.com/shows/28426",
              },
            },
            "externals": Object {
              "imdb": "tt4858114",
              "thetvdb": 328994,
              "tvrage": null,
            },
            "genres": Array [
              "Comedy",
              "Family",
              "Supernatural",
            ],
            "id": 28426,
            "image": Object {
              "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/119/297712.jpg",
              "original": "http://static.tvmaze.com/uploads/images/original_untouched/119/297712.jpg",
            },
            "language": "English",
            "name": "Hotel Transylvania: The Series",
            "network": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 78,
              "name": "Disney Channel",
            },
            "officialSite": "https://disneynow.com/shows/hotel-transylvania",
            "premiered": "2017-06-25",
            "rating": Object {
              "average": null,
            },
            "runtime": 15,
            "schedule": Object {
              "days": Array [
                "Saturday",
                "Sunday",
              ],
              "time": "08:30",
            },
            "status": "Running",
            "summary": "<p>Hotel Transylvania: The Series focuses on the teenage years of Dracula's daughter, Mavis, and her three closest friends Hank N Stein, Pedro and Wendy Blob. With Dracula away on \\"Official Vampire Business,\\" Mavis finally sees her chance to show her father what she can do. With the help of her friends, she decides to put her brilliant ideas of how to run the Hotel into motion, but enter Aunt Lydia, Dracula's older and much meaner sister. While Mavis and friends are ready to let loose and have fun, Aunt Lydia pushes for even more order and tradition at the Hotel.</p>",
            "type": "Animation",
            "updated": 1600986346,
            "url": "http://www.tvmaze.com/shows/28426/hotel-transylvania-the-series",
            "webChannel": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 83,
              "name": "DisneyNOW",
            },
            "weight": 96,
          },
          "summary": null,
          "type": "regular",
          "url": "http://www.tvmaze.com/episodes/1937214/hotel-transylvania-the-series-2x41-the-shawhank-redemption",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1937215"
    md={2}
    sm={4}
    xs={6}
  >
    <TvShowEpisodeCard
      onClick={[MockFunction]}
      scheduleItem={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/1937215",
            },
          },
          "airdate": "2020-10-25",
          "airstamp": "2020-10-25T12:45:00+00:00",
          "airtime": "08:45",
          "id": 1937215,
          "image": null,
          "name": "Friendship is Tragic",
          "number": 42,
          "runtime": 15,
          "season": 2,
          "show": Object {
            "_links": Object {
              "nextepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1937216",
              },
              "previousepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1937215",
              },
              "self": Object {
                "href": "http://api.tvmaze.com/shows/28426",
              },
            },
            "externals": Object {
              "imdb": "tt4858114",
              "thetvdb": 328994,
              "tvrage": null,
            },
            "genres": Array [
              "Comedy",
              "Family",
              "Supernatural",
            ],
            "id": 28426,
            "image": Object {
              "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/119/297712.jpg",
              "original": "http://static.tvmaze.com/uploads/images/original_untouched/119/297712.jpg",
            },
            "language": "English",
            "name": "Hotel Transylvania: The Series",
            "network": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 78,
              "name": "Disney Channel",
            },
            "officialSite": "https://disneynow.com/shows/hotel-transylvania",
            "premiered": "2017-06-25",
            "rating": Object {
              "average": null,
            },
            "runtime": 15,
            "schedule": Object {
              "days": Array [
                "Saturday",
                "Sunday",
              ],
              "time": "08:30",
            },
            "status": "Running",
            "summary": "<p>Hotel Transylvania: The Series focuses on the teenage years of Dracula's daughter, Mavis, and her three closest friends Hank N Stein, Pedro and Wendy Blob. With Dracula away on \\"Official Vampire Business,\\" Mavis finally sees her chance to show her father what she can do. With the help of her friends, she decides to put her brilliant ideas of how to run the Hotel into motion, but enter Aunt Lydia, Dracula's older and much meaner sister. While Mavis and friends are ready to let loose and have fun, Aunt Lydia pushes for even more order and tradition at the Hotel.</p>",
            "type": "Animation",
            "updated": 1600986346,
            "url": "http://www.tvmaze.com/shows/28426/hotel-transylvania-the-series",
            "webChannel": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 83,
              "name": "DisneyNOW",
            },
            "weight": 96,
          },
          "summary": null,
          "type": "regular",
          "url": "http://www.tvmaze.com/episodes/1937215/hotel-transylvania-the-series-2x42-friendship-is-tragic",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1954167"
    md={2}
    sm={4}
    xs={6}
  >
    <TvShowEpisodeCard
      onClick={[MockFunction]}
      scheduleItem={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/1954167",
            },
          },
          "airdate": "2020-10-25",
          "airstamp": "2020-10-25T13:00:00+00:00",
          "airtime": "09:00",
          "id": 1954167,
          "image": null,
          "name": "Episode 43",
          "number": 43,
          "runtime": 90,
          "season": 2020,
          "show": Object {
            "_links": Object {
              "nextepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1954168",
              },
              "previousepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1954167",
              },
              "self": Object {
                "href": "http://api.tvmaze.com/shows/15779",
              },
            },
            "externals": Object {
              "imdb": "tt0165001",
              "thetvdb": 264537,
              "tvrage": null,
            },
            "genres": Array [],
            "id": 15779,
            "image": Object {
              "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/237/592585.jpg",
              "original": "http://static.tvmaze.com/uploads/images/original_untouched/237/592585.jpg",
            },
            "language": "English",
            "name": "CBS News Sunday Morning",
            "network": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 2,
              "name": "CBS",
            },
            "officialSite": "http://www.cbsnews.com/sunday-morning/",
            "premiered": "1979-01-28",
            "rating": Object {
              "average": null,
            },
            "runtime": 90,
            "schedule": Object {
              "days": Array [
                "Sunday",
              ],
              "time": "09:00",
            },
            "status": "Running",
            "summary": "<p><b>CBS News Sunday Morning</b> is a early morning news and magazine program that airs on Sunday's on CBS.</p>",
            "type": "News",
            "updated": 1603327137,
            "url": "http://www.tvmaze.com/shows/15779/cbs-news-sunday-morning",
            "webChannel": null,
            "weight": 82,
          },
          "summary": null,
          "type": "regular",
          "url": "http://www.tvmaze.com/episodes/1954167/cbs-news-sunday-morning-2020-10-25-episode-43",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1949480"
    md={2}
    sm={4}
    xs={6}
  >
    <TvShowEpisodeCard
      onClick={[MockFunction]}
      scheduleItem={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/1949480",
            },
          },
          "airdate": "2020-10-25",
          "airstamp": "2020-10-25T14:00:00+00:00",
          "airtime": "10:00",
          "id": 1949480,
          "image": null,
          "name": "Episode 7",
          "number": 7,
          "runtime": 180,
          "season": 36,
          "show": Object {
            "_links": Object {
              "nextepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1953544",
              },
              "previousepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1949480",
              },
              "self": Object {
                "href": "http://api.tvmaze.com/shows/31986",
              },
            },
            "externals": Object {
              "imdb": "tt1239156",
              "thetvdb": null,
              "tvrage": null,
            },
            "genres": Array [
              "Sports",
            ],
            "id": 31986,
            "image": Object {
              "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/127/319991.jpg",
              "original": "http://static.tvmaze.com/uploads/images/original_untouched/127/319991.jpg",
            },
            "language": "English",
            "name": "Sunday NFL Countdown",
            "network": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 39,
              "name": "ESPN",
            },
            "officialSite": "http://espnmediazone.com/us/press-releases/2017/07/espn-announces-new-sunday-morning-fall-schedule/",
            "premiered": "1985-09-07",
            "rating": Object {
              "average": null,
            },
            "runtime": 180,
            "schedule": Object {
              "days": Array [
                "Sunday",
              ],
              "time": "10:00",
            },
            "status": "Running",
            "summary": "<p><b>Sunday NFL Countdown</b> is a sports talk show that airs Sunday mornings during Football season where a panel of NFL experts discuss and breakdown the upcoming weekend games. This show's former names include <i>NFL GameDay</i> from 1985 to 1995, <i>NFL Countdown</i> from 1996 to 1997, and since 1998, <i>Sunday NFL Countdown. </i>This Sunday pregame talf show debuted in 1985 and soon became a signature ESPN series behind host Chris Berman, who was at the helm for three decades. Taking over that role is Samantha Ponder, a one-time college football reporter and contributor. She introduces the features and directs the storylines associated with the day's games, joined by a roster of analysts highlighted by ex-NFL players. Chris Mortensen and Adam Schefter deliver breaking news and injury updates, while reporters at game sites set the scenes.</p>",
            "type": "Talk Show",
            "updated": 1603233206,
            "url": "http://www.tvmaze.com/shows/31986/sunday-nfl-countdown",
            "webChannel": null,
            "weight": 23,
          },
          "summary": "<p>Samantha Ponder, with a team of analysts, introduces the features and directs the storylines associated with the day's games.</p>",
          "type": "regular",
          "url": "http://www.tvmaze.com/episodes/1949480/sunday-nfl-countdown-36x07-episode-7",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1957581"
    md={2}
    sm={4}
    xs={6}
  >
    <TvShowEpisodeCard
      onClick={[MockFunction]}
      scheduleItem={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/1957581",
            },
          },
          "airdate": "2020-10-25",
          "airstamp": "2020-10-25T14:00:00+00:00",
          "airtime": "10:00",
          "id": 1957581,
          "image": null,
          "name": "Mo and Bo Hit the Road",
          "number": 29,
          "runtime": 15,
          "season": 2,
          "show": Object {
            "_links": Object {
              "previousepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1957582",
              },
              "self": Object {
                "href": "http://api.tvmaze.com/shows/40260",
              },
            },
            "externals": Object {
              "imdb": "tt9433014",
              "thetvdb": 356827,
              "tvrage": null,
            },
            "genres": Array [
              "Comedy",
              "Adventure",
              "Children",
            ],
            "id": 40260,
            "image": Object {
              "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/179/447727.jpg",
              "original": "http://static.tvmaze.com/uploads/images/original_untouched/179/447727.jpg",
            },
            "language": "English",
            "name": "Abby Hatcher",
            "network": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 206,
              "name": "Nick Jr.",
            },
            "officialSite": "http://www.nickjr.com",
            "premiered": "2019-01-01",
            "rating": Object {
              "average": null,
            },
            "runtime": 15,
            "schedule": Object {
              "days": Array [
                "Sunday",
              ],
              "time": "10:00",
            },
            "status": "Running",
            "summary": "<p>Seven-year-old Abby Hatcher and her Fuzzly friend Bozzly go on adventures to fix the Fuzzly's problems at her hotel.</p>",
            "type": "Animation",
            "updated": 1603663609,
            "url": "http://www.tvmaze.com/shows/40260/abby-hatcher",
            "webChannel": null,
            "weight": 71,
          },
          "summary": "<p>Mo and Bo are off to see huge tap shoes at a park, but they only have so many minutes before it is closed.</p>",
          "type": "regular",
          "url": "http://www.tvmaze.com/episodes/1957581/abby-hatcher-2x29-mo-and-bo-hit-the-road",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1957582"
    md={2}
    sm={4}
    xs={6}
  >
    <TvShowEpisodeCard
      onClick={[MockFunction]}
      scheduleItem={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/1957582",
            },
          },
          "airdate": "2020-10-25",
          "airstamp": "2020-10-25T14:15:00+00:00",
          "airtime": "10:15",
          "id": 1957582,
          "image": null,
          "name": "Happy Fuzzly Birthday",
          "number": 30,
          "runtime": 15,
          "season": 2,
          "show": Object {
            "_links": Object {
              "previousepisode": Object {
                "href": "http://api.tvmaze.com/episodes/1957582",
              },
              "self": Object {
                "href": "http://api.tvmaze.com/shows/40260",
              },
            },
            "externals": Object {
              "imdb": "tt9433014",
              "thetvdb": 356827,
              "tvrage": null,
            },
            "genres": Array [
              "Comedy",
              "Adventure",
              "Children",
            ],
            "id": 40260,
            "image": Object {
              "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/179/447727.jpg",
              "original": "http://static.tvmaze.com/uploads/images/original_untouched/179/447727.jpg",
            },
            "language": "English",
            "name": "Abby Hatcher",
            "network": Object {
              "country": Object {
                "code": "US",
                "name": "United States",
                "timezone": "America/New_York",
              },
              "id": 206,
              "name": "Nick Jr.",
            },
            "officialSite": "http://www.nickjr.com",
            "premiered": "2019-01-01",
            "rating": Object {
              "average": null,
            },
            "runtime": 15,
            "schedule": Object {
              "days": Array [
                "Sunday",
              ],
              "time": "10:00",
            },
            "status": "Running",
            "summary": "<p>Seven-year-old Abby Hatcher and her Fuzzly friend Bozzly go on adventures to fix the Fuzzly's problems at her hotel.</p>",
            "type": "Animation",
            "updated": 1603663609,
            "url": "http://www.tvmaze.com/shows/40260/abby-hatcher",
            "webChannel": null,
            "weight": 71,
          },
          "summary": "<p>The Fuzzlies want to offer gifts for Abby's birthday party, but they have differing ideas which get into each other's way.</p>",
          "type": "regular",
          "url": "http://www.tvmaze.com/episodes/1957582/abby-hatcher-2x30-happy-fuzzly-birthday",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    justify="center"
    xs={12}
  >
    <WithStyles(ForwardRef(Link))
      className="makeStyles-showMore-1"
      component="button"
      onClick={[Function]}
      variant="body2"
    >
      Show more
    </WithStyles(ForwardRef(Link))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
